new-packages/Core/src/Result.candy

```diff
@@ -19,3 +19,15 @@ impl[Success, ErrorType] Result[Success, ErrorType] {
     )
   }
 }
+
+impl[Success: Equals, ErrorType: Equals] Result[Success, ErrorType]: Equals {
+  fun equals(other: Result[Success, ErrorType]): Bool {
+    this.match[Bool](
+      ok = { thisSuccess -> other.match[Bool](ok = { thisSuccess == it }, error = { false() }) },
+      error = { thisError -> other.match[Bool](ok = { false() }, error = { thisError == it }) },
+    )
+  }
+}
+impl[Success: Hash, ErrorType: Hash] Result[Success, ErrorType]: Hash {
+  fun hash[H: Hasher](): H { this.match[H](ok = { it.hash[H]() }, error = { it.hash[H]() }) }
```

Comment:
If `Result Success` and `Result Error` contain the same value, their hashes would be identical. Maybe we should also `combine` this with a constant representing the case?